diff --git a/src/dnsmasq.c b/src/dnsmasq.c
index 112dbe5..10257cf 100644
--- a/src/dnsmasq.c
+++ b/src/dnsmasq.c
@@ -162,7 +162,8 @@ int main (int argc, char **argv)
     open("/dev/null", O_RDWR); 
   
   /* Close any file descriptors we inherited apart from std{in|out|err} */
-  close_fds(max_fd, -1, -1, -1);
+  // close_fds(max_fd, -1, -1, -1);
+  close_fds(1024, -1, -1, -1);
   
 #ifndef HAVE_LINUX_NETWORK
 #  if !(defined(IP_RECVDSTADDR) && defined(IP_RECVIF) && defined(IP_SENDSRCADDR))
diff --git a/src/util.c b/src/util.c
index 6521e3f..b562422 100644
--- a/src/util.c
+++ b/src/util.c
@@ -43,7 +43,7 @@ static int outleft = 0;
 
 void rand_init()
 {
-  int fd = open(RANDFILE, O_RDONLY);
+  /* int fd = open(RANDFILE, O_RDONLY);
   
   if (fd == -1 ||
       !read_write(fd, (unsigned char *)&seed, sizeof(seed), 1) ||
@@ -51,6 +51,9 @@ void rand_init()
     die(_("failed to seed the random number generator: %s"), NULL, EC_MISC);
   
   close(fd);
+  */
+  memset(seed, 0, sizeof(seed));
+  memset(in, 0, sizeof(in));
 }
 
 #define ROTATE(x,b) (((x) << (b)) | ((x) >> (32 - (b))))
